### DISCUSSION TITLE
ScriptReference/SerializedProperty.Next

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#46] [Notes]
# ScriptReference/SerializedProperty.Next

https://docs.unity3d.com/ScriptReference/SerializedProperty.Next.html

**Comment by vertxxyz** [comment]
To iterate over the children of a Property as mentioned in [`GetEndProperty`](https://docs.unity3d.com/ScriptReference/SerializedProperty.GetEndProperty.html), you need to:
```csharp
SerializedProperty end = prop.GetEndProperty();
while (prop.Next(true) && !SerializedProperty.EqualContents(prop, end))
{
    ...
}
```
Where prop is your SerializedProperty.
Using the `Next` iterator will advance your SerializedProperty, so use [`Copy`](https://docs.unity3d.com/ScriptReference/SerializedProperty.Copy.html) if you want to make a copy to iterate over.
